sdk/core/azure-core/src/main/java/com/azure/core/util/ClientOptions.java

```diff
@@ -73,6 +74,9 @@ public ClientOptions setHeaders(Iterable<Header> headers) {
      * @return the headers.
      */
     public Iterable<Header> getHeaders() {
+        if (headers == null) {
+            headers = new ArrayList<>();
```

Comment:
Rather than return a new arraylist - just return an empty iterable, e.g. `return Collections.emptyList();`

Reply:
Yeah, this creates a new instance of ArrayList each time. Instead, using Collections.emptyList() will return a static singleton list instance.